snuba/datasets/processors/generic_metrics_processor.py

```diff
@@ -146,10 +158,20 @@ def process_message(
                 message["sentry_received_timestamp"]
             )
 
+        self.__record_cogs(message)
         return InsertBatch(
             [processed], None, sentry_received_timestamp=sentry_received_timestamp
         )
 
+    def __record_cogs(self, message: GenericMetric) -> None:
+        if random() < (get_config("gen_metrics_processor_cogs_probability") or 0):
```

Comment:
Curious, why do you sample this?

Reply:
This is just to have a gradual rollout, it won't be sampled in its final state.

Reply:
You should buffer the counts in memory if you want to produce less messages. Then you don't need a gradual rollout or to produce one message per cogs item recorded in the future